include/ginkgo/core/distributed/assembly_helpers.hpp

```diff
@@ -0,0 +1,57 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#ifndef GKO_PUBLIC_CORE_DISTRIBUTED_ASSEMBLY_HELPERS_HPP_
```

Comment:
I would consider renaming assembly_helpers to just assembly, since the functions essentially do the full distributed part of the assembly, not just help in it.